### PR TITLE
Bug 1124816 - Focus window if it is top most +autoland

### PR DESCRIPTION
**Pull request by alivedise**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[alivedise Alive.Kuo (alivedise)](https://github.com/alivedise) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=fcd8ed6cd2bb)

File: apps/system/js/app_transition_controller.js [hunk 1]
```diff
@@ -292,9 +292,19 @@
...
-            !rocketbar.active &&
-            !Service.query('SimLockManager.isActive'));
+            Service.query('getTopMostWindow') === this.app &&
+            Service.query('getTopMostUI').name.indexOf(bottomWindowType) >= 0);
+    // Possible bottom most windows:
+    // AppWindow/HomscreenWindow, AttentionWindow, SecureWindow,
+    // LockScreenWindow
+    // which belong to
+    // AppWindowManager, AttentionWindowManager, SecureWindowManager,
+    // LockScreenWindowManager.
```
Comment (etiennesegonzac) [inline apps/system/js/app_transition_controller.js]: Ok. This makes sense but I'd like to see if we can come up with a stronger alternative than an `str.startsWith`...

* maybe having a `const` mapping `CLASS_NAME`s to hierarchy UI names
* or giving each window instance a reference to its `.manager` (but I could see why we don't want to go there...)

What do you think?

Comment (alivedise) [inline apps/system/js/app_transition_controller.js]: I prefer the first one but I am thinking where to put it..

Comment (alivedise) [inline apps/system/js/app_transition_controller.js]: What if <code>AppWindow.prototype.MANAGER = 'AppWindowManager';</code> ?

**Comment by try-server-hook** [comment]
[alivedise Alive.Kuo (alivedise)](https://github.com/alivedise) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=23173cd05017)

File: apps/system/js/app_transition_controller.js [hunk 2]
```diff
@@ -292,9 +292,11 @@
     // See: https://bugzilla.mozilla.org/show_bug.cgi?id=961557
     // XXX: We should let HierarchyManager to manage the focus.
     // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1079748
+    var bottomWindow = this.app.getBottomMostWindow();
     return (this._transitionState == 'opened' &&
-            !rocketbar.active &&
-            !Service.query('SimLockManager.isActive'));
+            Service.query('getTopMostWindow') === this.app &&
+            Service.query('getTopMostUI').name ===
+            bottomWindow.HIERARCHY_MANAGER);
```
Comment (etiennesegonzac) [inline apps/system/js/app_transition_controller.js]: :+1:

**Pull request merged**